### PR TITLE
SLING-8251 - Support checking dependencies for content packages

### PR DESCRIPTION
**Pull request by simonetripodi**

### CODE REVIEW

**Comment by karlpauls** [comment]
@simonetripodi, i think we shouldn't introduce a ShutdownHook for deleting the tmp files. If we want to be nice, just call deleteOnExit() on all the files created but don't try to delete them yourself via ShutdownHook.

**Comment by simonetripodi** [comment]
@karlpauls done

**Comment by simonetripodi** [comment]
@cziegeler @bosschaert please note the checks fail due to missing recently released artifacts not yet on central repo

**Comment by karlpauls** [comment]
@simonetripodi, that is problematic as deleteOnExit is not going to delete dirs if they are not empty. I was more thinking about not removing the try/finally but rather change the deleteRecursive to look like:

```
private void deleteOnExitRecursive(File file) {
	        if (file.isDirectory()) {
	            File[] childs = file.listFiles();
	            if (childs != null) {
	                for (File child : childs) {
	                    deleteRecursive(child);
	                }
	            }
	        }
                file.deleteOnExit();
}
```

**Comment by simonetripodi** [comment]
@karlpauls nice!

**Comment by simonetripodi** [comment]
@karlpauls included!

File: src/main/java/org/apache/sling/feature/analyser/task/impl/CheckContentPackagesDependencies.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+        }
+
+        for (PackageId contentPackageId : dependenciesMap.keySet()) {
+            verifyDependenciesTree(ctx, contentPackageId, dependenciesMap);
+        }
+    }
+
+    private void onDescriptor(AnalyserTaskContext ctx, ArtifactDescriptor descriptor, Map<PackageId, Dependency[]> dependenciesMap) throws Exception {
+        URL resourceUrl = descriptor.getArtifactFile();
+        String filename = resourceUrl.getFile().replace('/', File.separatorChar);
```
Comment (karlpauls) [inline src/main/java/org/apache/sling/feature/analyser/task/impl/CheckContentPackagesDependencies.java]: @simonetripodi, not sure that is the best approach. You can use o.a.s.feature.io.IOUtils to get the File from the url.

Comment (simonetripodi) [reply src/main/java/org/apache/sling/feature/analyser/task/impl/CheckContentPackagesDependencies.java]: That's where I copied it :P

Comment (karlpauls) [reply src/main/java/org/apache/sling/feature/analyser/task/impl/CheckContentPackagesDependencies.java]: Should be fixed there as well - maybe just do it in this PR too.

Comment (simonetripodi) [reply src/main/java/org/apache/sling/feature/analyser/task/impl/CheckContentPackagesDependencies.java]: I didn't realise we had common-io already in the dependencies set, I am replacing the homemade method with Apache one

Comment (simonetripodi) [reply src/main/java/org/apache/sling/feature/analyser/task/impl/CheckContentPackagesDependencies.java]: done

Comment (karlpauls) [reply src/main/java/org/apache/sling/feature/analyser/task/impl/CheckContentPackagesDependencies.java]: not the commons-io - the one from feature.io

Comment (simonetripodi) [reply src/main/java/org/apache/sling/feature/analyser/task/impl/CheckContentPackagesDependencies.java]: done

**Comment by karlpauls** [comment]
lgtm - Thanks!

**Pull request merged**